src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -62,8 +63,8 @@ private synchronized void loadExecutions() {
                 }
             }
             if (execution != null) {
-                // TODO JENKINS-37154 we would rather not block here
-                for (StepExecution se : execution.getCurrentExecutions(true).get()) {
+                // JENKINS-37154 sometimes we must block here in order to get accurate results
+                for (StepExecution se : execution.getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
> The system might be heavily loaded and just need a little time. The script might be doing some calculation in the CPS VM thread. I would not want to cause the list of inputs to be incorrect merely because of some transient load.

I can see it needing a bit of time, yes --  basically execution time for one (non-asynchronous) step, right?  Or am I missing something?  Feels like if that's taking more than a few seconds, it represents a bit of an issue (should be converted to an async step) and we're better off just timing out in that case.  

My concern is that generally there are a limited (not tiny, but still limited) number of request handler threads, and this completely locks one while waiting.  A full minute is awfully long to be holding a thread that is locked up potentially -- better than infinity, but we can still get thread exhaustion.